<html>
<head>
<meta charset='utf-8'>
<style>
.pass {
    font-weight: bold;
    color: green;
}
.fail {
    font-weight: bold;
    color: red;
}
</style>

<script>
if (window.testRunner)
    testRunner.dumpAsText();

function SputnikError(message)
{
    this.message = message;
}

SputnikError.prototype.toString = function ()
{
    return 'SputnikError: ' + this.message;
};

var sputnikException;

function testPrint(msg)
{
    var span = document.createElement("span");
    document.getElementById("console").appendChild(span); // insert it first so XHTML knows the namespace 
    span.innerHTML = msg + '<br />';
}

function escapeHTML(text)
{
    return text.toString().replace(/&/g, "&amp;").replace(/</g, "&lt;");
}

function printTestPassed(msg)
{
    testPrint('<span><span class="pass">PASS</span> ' + escapeHTML(msg) + '</span>');
}

function printTestFailed(msg)
{
    testPrint('<span><span class="fail">FAIL</span> ' + escapeHTML(msg) + '</span>');
}

function testFailed(msg)
{
    throw new SputnikError(msg);
}

var successfullyParsed = false;
</script>

</head>
<body>
<p>S12.14_A6</p>
<div id='console'></div>
<script>
try {

/**
 * @name: S12.14_A6;
 * @section: 12.14;
 * @assertion:  The production TryStatement: "try Block Catch Finally";
 * @description: Executing sequence of "try" statements, using counters with varying values within;
 */

// CHECK#1
var c1=0;
try {
  c1+=1;
  y;
  testFailed('#1.1: "y" lead to throwing exception');
}
catch (e) {
  c1*=2;
}
if (c1!==2){
  testFailed('#1.2: Sequence evaluation of commands try/catch is 1. try, 2. catch');	
}

// CHECK#2
var c2=0;
try{
  c2+=1;
}
finally{
  c2*=2;
}
if (c2!==2){
  testFailed('#2: Sequence evaluation of commands try/finally is 1. try, 2. finally');
}

// CHECK#3
var c3=0;
try{
  c3=1;
  z;
}
catch(err){
  c3*=2;
}
finally{
  c3+=1;
}
if (c3!==3){
  testFailed('#3: Sequence evaluation of commands try/catch/finally(with exception) is 1. try, 2. catch, 3. finally');
}	

// CHECK#4
var c4=0;
try{
  c4=1;
}
catch(err){
  c4*=3;
}
finally{
  c4+=1;
}
if (c4!==2){
  testFailed('#4: Sequence evaluation of commands try/catch/finally(without exception) is 1. try, 2. finally');
}	

} catch (ex) {
    sputnikException = ex;
}

var successfullyParsed = true;
</script>

<script>
if (!successfullyParsed)
    printTestFailed('successfullyParsed is not set');
else if (sputnikException)
    printTestFailed(sputnikException);
else
    printTestPassed("");
testPrint('<br /><span class="pass">TEST COMPLETE</span>');
</script>
</body>
</html>
